nodes/src/nodes/agent_crewai/crewai_manager/manager.py

```
@@ -48,6 +48,48 @@
+    #    tool observation (no Final Answer written).  Only apply when the text
+    #    looks like a ReAct trace; otherwise clean output (e.g. ```chartjs
+    #    fences) would have their prefix incorrectly stripped.
+    #
+    #    Require the trailing JSON to extend to end-of-text (last non-whitespace
+    #    char is `}` or `]`).  Without this guard, an intermediate Observation
+    #    containing JSON followed by more Thought/Action lines would match via
+    #    rfind('\n{') even though the true final answer is later free text.
+    react_markers = ('Thought:', 'Action:', 'Observation:')
+    if any(m in text for m in react_markers):
+        tail = text.rstrip()
+        if tail.endswith('}') or tail.endswith(']'):
```

Ruff (0.15.9)

[warning] Call `endswith` once with a `tuple`

Merge into a single `endswith` call

(PIE810)